### PR TITLE
[Declaration redundancy] Actual method parameter is the same constant - tests

### PR DESCRIPTION
**Pull request by abhipsaMisra**
The warning has been suppressed in test classes, since the flagged params were being referenced in helper methods

### CODE REVIEW

File: deps/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/deps/serializer/MethodParserTest.java [hunk 1]
```diff
@@ -579,7 +583,7 @@ private static final TestMethod createMethodRequestWithConnectTimeout(
         return testMethod;
     }
 
-    private static final TestMethod createMethodResponse(Integer status, String payload, boolean wrapPayloadAsString, Object expectedPayload)
```
Comment (abhipsaMisra) [inline deps/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/deps/serializer/MethodParserTest.java:582]: no point in declaring private static methods as final, they cannot be overridden anyway

File: iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/setup/DeviceTwinCommon.java [hunk 2]
```diff
@@ -512,7 +513,7 @@ public void tearDownNewDeviceAndModule()
         }
     }
 
-    protected void readReportedPropertiesAndVerify(DeviceState deviceState, String startsWithKey, String startsWithValue, int expectedReportedPropCount) throws IOException, IotHubException, InterruptedException
```
Comment (abhipsaMisra) [inline iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/setup/DeviceTwinCommon.java:515]: the property key is a constant, so replaced it inline

**Review by timtay-microsoft — APPROVED** [review]

**Pull request merged**